### DISCUSSION TITLE
Android build on real device crashes (BEGINNER)

### DISCUSSION THREAD

**Discussion by aadiby** [quasarframework/quasar#9804] [CLI - Capacitor / Cordova modes]
Hi All,

This is my very first quasar and android app. i am following the documents to the letter and it all works except last step:
- app developed on linux
- android emulator build works fine (except no internet connection for app on emulator for some reason?)
- quasar build -m cordova -T android, signed and zipaligned using this documentation: https://quasar.dev/quasar-cli/developing-cordova-apps/publishing-to-store
- installed APK on own device (not published to store), no issue
- When i try to **open the app on real device, it simply crashes... no errors, nothing**

How would i go about to understanding why the app is crashing on my real android device while it works on emulator? (and perhaps fixing app internet connection on emulator)

**Comment by botura** [comment]
I don´t have the answer for your question. But just curious why are you using cordova instead of capacitor?

Are you using chrome tools to debug your app on the android device? `chrome://inspect/#devices` on chrome address bar.

**Comment by metalsadman** [comment]
use `--debug` flag on your build command ie. `quasar build -m android --debug`. open your IDE, open logcat, run your app, and check for error logs.
